src/segment_regquery_windows.go

```diff
@@ -0,0 +1,143 @@
+//go:build windows
+
+// TODO:
+//
+//  * reg value -> string formatting to own fn?
+//  * support for other reg types?
+//  * custom formatting for values?  (custom sprintf is too dangerous, what else?)
+//
+
+package main
+
+import (
+	"fmt"
+	"strings"
+	"unsafe"
+
+	"golang.org/x/sys/windows"
+)
+
+func (r *regquery) getHKEYValueFromAbbrString(abbr string) windows.Handle {
+	var ret windows.Handle = 0
+
+	switch abbr {
+	case "HKCR", "HKEY_CLASSES_ROOT":
+		ret = windows.HKEY_CLASSES_ROOT
+	case "HKCC", "HKEY_CURRENT_CONFIG":
+		ret = windows.HKEY_CURRENT_CONFIG
+	case "HKCU", "HKEY_CURRENT_USER":
+		ret = windows.HKEY_CURRENT_USER
+	case "HKLM", "HKEY_LOCAL_MACHINE":
+		ret = windows.HKEY_LOCAL_MACHINE
+	case "HKU", "HKEY_USERS":
+		ret = windows.HKEY_USERS
+	}
+
+	return ret
+}
+
+//
+// Takes a registry path like "HKLM\Software\Microsoft\Windows NT\CurrentVersion" and a key under that path like "CurrentVersion" (or "" if the (Default) key is required).
+// Returns a bool and string:
+//
+//   true and the retrieved value formatted into a string if successful.
+//   false and the string will be the error
+//
+func (r *regquery) getRegistryKeyValue(regPath string, regKey string) (bool, string) {
+
+	// using short-circuit logic as previous nested if version was horrible to understand.
+
+	// Extract root HK value and turn it into a windows.Handle to open the key.
+	regPathParts := strings.SplitN(regPath, "\\", 2)
+
+	regRootHKeyHandle := r.getHKEYValueFromAbbrString(regPathParts[0])
+	if regRootHKeyHandle == 0 {
+		return false, fmt.Sprintf("Error, Supplied root HKEY value not valid: '%s'", regPathParts[0])
```

Comment:
in `environment_windows.go`, you can make use of `env.log(Error, "getRegistryKeyValue", err.Error())`

Reply:
splendid - logging and trace code added.